frontend/src/metabase/admin/performance/hooks/useConfirmIfFormIsDirty.tsx

```diff
@@ -0,0 +1,44 @@
+import { useState, useCallback } from "react";
+import type { InjectedRouter, Route } from "react-router";
+import { t } from "ttag";
+
+import useBeforeUnload from "metabase/hooks/use-before-unload";
+import { useConfirmation } from "metabase/hooks/use-confirmation";
+
+import { useConfirmOnRouteLeave } from "./useConfirmOnRouteLeave";
+
+export const useConfirmIfFormIsDirty = (
+  router: InjectedRouter,
+  route?: Route,
+) => {
+  const [isStrategyFormDirty, setIsStrategyFormDirty] = useState(false);
+
+  const { show: askConfirmation, modalContent: confirmationModal } =
+    useConfirmation();
+
+  const askBeforeDiscardingChanges = useCallback(
+    (onConfirm: () => void) =>
+      askConfirmation({
+        title: t`Discard your changes?`,
+        message: t`Your changes haven’t been saved, so you’ll lose them if you navigate away.`,
+        confirmButtonText: t`Discard`,
+        onConfirm,
+      }),
+    [askConfirmation],
+  );
+
+  useConfirmOnRouteLeave({
+    router,
+    route,
```

Comment:
can this hook get the router information with a selector? that would keep us from having to use the `withRouter` hoc and pass all these through various hooks

Reply:
Looking through the codebase, I don't think this can be done, but please lmk if I am wrong